flocker/acceptance/testtools.py

```diff
@@ -911,6 +911,13 @@ def detach_and_destroy_volume(volume):
                 d.addBoth(
                     lambda _: async_api.destroy_volume(volume.blockdevice_id)
                 )
+                # Consume failures and write them out. Failures might just
+                # indicate that the cluster beat us to deleting the volume
+                # which should not cause the test to fail, we only want the
+                # test to fail if list_volumes still returns a non-empty list.
+                # The construction of api_clean_state should prevent this from
+                # masking significant failures to clean up volumes.
+                d.addErrback(write_failure)
```

Comment:
Ah, yea this is relatively new (I think it was added in the past month, perhaps as part of the acceptance test for the AWS duplicate dataset bug?). I think this might be occurring in GCE more than in other drivers because GCE sometimes has some very slow operations, but I think we now do this for all backends.

It _might_ have something to do with the dataset flickering over the flocker API though....
Sounds like people have seen that on non-GCE though.

I believe:

- This cleanup code is new, and not yet hardened (although it runs on all backends).
- It may or may not have similar failure modes on EBS and Cinder, but it depends on a lot on the details of how the endpoint API behaves while being queried from multiple nodes about the same dataset, and I'm not sure if I see too much value in looking at those details in the short term.
- Catching errors here is still correct, as we should never get out of this cleanup method until list_volumes on the API returns an empty list.
- This change hardens the cleanup code in all cases.